rpc/trace.go

```diff
@@ -16,3 +17,29 @@ func (provider *Provider) TransactionTrace(ctx context.Context, hash string) err
 func (provider *Provider) TraceBlockTransactions(ctx context.Context, hash string) error {
 	return ErrNotImplemented
 }
+
+// simulate a given transaction on the requested state, and generate the execution trace
+func (provider *Provider) SimulateTransactions(
+	ctx context.Context,
+	blockId BlockID,
+	txns []BroadcastedTransaction,
+	simulationFlags []SimulationFlag) ([]SimulatedTransaction, error) {
```

Comment:
Could we leave this in just one line? I think it would be best to be consistent with the rest of the code.

We could use a tool for code formatting and add it to the workflow, so it makes sure the code is uniformly formatted, wyt?